peps/pep-0815.rst

```diff
@@ -0,0 +1,2282 @@
+names (``mypackage-gpu``, ``mypackage-cpu``, etc.). Each of these approaches
+has significant drawbacks, such as excessive binary size, dependency
+confusion, inefficient dependency resolution, complex documentation,
+etc.
+
+According to the `2024 Python Developers
+Survey <https://lp.jetbrains.com/python-developers-survey-2024/#purposes-for-using-python>`__,
+a significant portion of respondents over the last years have been
+successively using Python for scientific computing purposes, covering
+such areas as Data analysis (steadily over 40% respondents), Machine
+learning (grown to 40% in 2024), Data engineering (around 30%), and
+more. Many of these use cases are directly impacted by suboptimal
+packaging.
+
+This issue is often crossing the boundaries of scientific computing - as
+highlighted in the following issue: `manylinux_2_34 x86-64 builds
+produce binaries that are not compatible with all x86-64
+CPUs <https://github.com/pypa/manylinux/issues/1725>`__, where
+``manylinux_2_34_x86_64`` now implicitly requires ``x86_64-v2`` with no
+support for other x86 version. This lack of support results in
+complexity in managing platform-specific dependencies and compatibility.
+That complexity affects the installation process for users and increases
+the maintenance burden for package authors. The ``x86_64`` compiler
+flags further emphasizes the urgent need for a more expressive and
+efficient solution in the Python packaging ecosystem.
+
+As illustrated by
+`archspec <https://github.com/archspec/archspec>`__’s authors and
+maintainers, the ability to optimize a package for a specific
+architecture can lead to significant performance improvement (and often
+also reduce power requirements, etc.).
+
+.. figure:: pep-0815/avx512_gromacs_benchmark.png
+    :alt: A bar graph comparing GROMACS performance (in ns/day) with
+          various targets. The first two bars are labeled "yum (2018.8)"
+          and "generic (SSE2)", reach about 1.0 ns/day and are both
+          marked as "SSE2". The next bar is labeled "ivybridge" ("AVX")
+          and reaches almost 1.5 ns/day.  Two following bars are labeled
+          "haswell" and "broadwell" (both "AVX2") and exceed 1.5 ns/day
+          slightly.  The last two bars are labeled "skylake_avx512" and
+          "cascadelake" (both "AVX512") and reach almost 2.0 ns/day.
+
+    GROMACS AVX512 Benchmark
+
+..
+
+    “Compiling `GROMACS <https://www.gromacs.org/>`__ for architectures
+    that can exploit the AVX-512 instructions supported by the Intel
+    Cascade Lake microarchitecture gives an additional 18% performance
+    improvement relative to using AVX2 instructions, with a speedup of
+    about 70% compared to a generic GROMACS installation with only SSE2.”
+
+    — `archspec: A library for detecting, labeling, and reasoning about
+    microarchitectures <https://tgamblin.github.io/pubs/archspec-canopie-hpc-2020.pdf>`__
+
+This PEP proposes a systematic and scalable approach to selecting
```

Comment:
I'd reduce those sentences (e.g. omit this paragraph), I think the use cases and quotes speak for themselves and make a stronger point.